server/src/controllers/userController.ts

```diff
@@ -26,27 +33,33 @@ const userDetails = async (req: Request, res: Response, next: NextFunction) => {
 const updateUserDetails = async (req: Request, res: Response, next: NextFunction) => {
     let cloudnaryResult: UploadedImage[] = []
     try {
-        const { userId } = req.user;
+        const { userId, currentPassword, newPassword, isPasswordChange } = req.body;
         const imageFile = req.files?.profileImage;
 
+        const updatingData = {
+            ...req.body
+        }
+
         const userDetails = await getUserDetails(userId, next);
         if (!userDetails) {
             return next(new NotFoundError("User not found"))
         }
 
+        if (isPasswordChange == "true") {
```

Comment:
why using string 'true' 
use only  this type of condition` if (isPasswordChange)`